### PR TITLE
changed the sizing logic for ObjectPool

### PR DESCRIPTION
**Pull request by cwiese**
The PR fixes the datatype selection logic for ObjectPool.   Currently, a file if 255 row with 255 unique string values will fail as it attempts to use UInt8.   

Reproduce by "readsas" the attached file.

PS) finally fixed a bug from Tom :-)

### CODE REVIEW

**Comment by cwiese** [comment]
[pooltest255.sas7bdat.zip](https://github.com/tk3369/SASLib.jl/files/6213390/pooltest255.sas7bdat.zip)

File: src/SASLib.jl [hunk 1]
```diff
@@ -912,11 +912,11 @@ function createstrarray(handler, column_symbol, nrows)
     elseif haskey(handler.config.string_array_fn, :_all_)
         handler.config.string_array_fn[:_all_](nrows)
     else
-        if nrows <= 2 << 7 
+        if nrows + 1 < 2 << 7
```
Comment (tk3369) [inline src/SASLib.jl:915]: Do you need to add 1 here since you've already changed the operator to `<`?

**Review by tk3369 — COMMENTED** [review]
Thanks for the contribution 👍🏼 

Can you also:
1. update add the test file and update the test
2. bump version in Project.toml

File: src/SASLib.jl (hunk 1 above)
Comment (cwiese) [reply src/SASLib.jl:915]: yes - it was deliberate

Comment (cwiese) [reply src/SASLib.jl:915]: yes.  both change were required .. I can add a file with 1 row, 254 rows and 256 row also if you want?

Comment (cwiese) [reply src/SASLib.jl:915]: @tk3369 is this Test sufficient?   I feel strongly it correct - we have production issues :-)

**Review by tk3369 — APPROVED** [review]

**Comment by tk3369** [comment]
Thank you @cwiese !

**Pull request merged**